pkgs/applications/editors/emacs/macport.nix

```diff
@@ -27,13 +27,26 @@ stdenv.mkDerivation rec {
     sha256 = "0f2wzdw2a3ac581322b2y79rlj3c9f33ddrq9allj97r1si6v5xk";
   };
 
+  patches = [
+    (fetchpatch {
+      url = "https://raw.githubusercontent.com/railwaycat/homebrew-emacsmacport/7e793808ebbc11d519a0103fb9f8fe7efbec345d/patches/mac-arm-fix.diff";
+      sha256 = "sha256-RF9b5PojFUAjh2TDUW4+HaWveV30Spy1iAXhaWf1ZVg=";
+    })
+  ];
+
   enableParallelBuilding = true;
 
-  nativeBuildInputs = [ pkg-config autoconf automake ];
+  nativeBuildInputs = [
+    pkg-config autoconf automake
+  ] ++ lib.optionals stdenv.isAarch64 [
+    sigtool
+  ];
```

Comment:
Why only on aarch64?

Reply:
It's been a while, so I'm unable to give a proper answer. But something to do with ARM macOS requiring a code signature, while Intel macOS does not.

Reply:
IIRC emacs build system does not invoke sigtool on x86_64-darwin at all, so I didn't find it necessary to declare it as a dependency for it. Doesn't hurt to remove conditional, I just go by declaring bare minimum required to build & run the thing successfully.